parsers/AR.py

```diff
@@ -0,0 +1,563 @@
+
+    soup = BeautifulSoup(req.content, 'html.parser')
+    figs = soup.find_all("div", class_="r11")
+    data_table = [str(tag.get_text()) for tag in figs]
+
+    return data_table
+
+
+def get_datetime():
+    """
+    Generation data is updated hourly.  Makes request then finds most recent hour available.
+    Returns an arrow datetime object using UTC-3 for timezone and zero for minutes and seconds.
+    """
+
+    #Argentina does not currently observe daylight savings time.  This may change from year to year!
+    #https://en.wikipedia.org/wiki/Time_in_Argentina
+    rt = requests.get(url)
+    timesoup = BeautifulSoup(rt.content, 'html.parser')
+    find_hour = timesoup.find("option", selected = "selected", value = "1" ).getText()
+    at = arrow.now('UTC-3')
+    datetime = (at.replace(hour = int(find_hour), minute = 0, second = 0)).format('YYYY-MM-DD HH:mm:ss')
+
+    return {'datetime': datetime}
+
+
+def dataformat(junk):
+    """Takes string data with only digits and returns it as a float."""
+
+    formatted=[]
+    for item in junk:
+        if not any(char in item for char in string.ascii_letters):
+            item = float(item.replace(',','.'))
+        formatted.append(item)       
+
+    return formatted
+
+
+def get_thermal():
+    """
+    Requests thermal generation data then parses and sorts by type.  Nuclear is included.
+    Returns a dictionary.
+    """
+
+    #Need to persist session in order to get ControlID and ReportSession so we can send second request
+    #for table data.  Both these variables change on each new request.
+    s = requests.Session()
+    r = s.get(url)
+    pat = re.search("ControlID=[^&]*", r.text).group()
+    spat = re.search("ReportSession=[^&]*", r.text).group()
+    cid = pat.rpartition('=')[2]
+    rs = spat.rpartition('=')[2]
+    full_table = []
+
+    #'En Reserva' plants are not generating and can be ignored.
+
+    for pagenumber in range(1,8,1):
+        t = s.get(turl, params = {'ControlID': cid, 'ReportSession': rs, 'PageNumber': '{}'.format(pagenumber)})
+        text_only = webparser(t)
+        if 'En Reserva' in text_only:
+            break
+        full_table.append(text_only)
+
+    data = list(itertools.chain.from_iterable(full_table))
+    formatted_data = dataformat(data)
+    mapped_data = [power_plant_type.get(x,x) for x in formatted_data]
+    
+    find_totals = [i+1 for i,x in enumerate(mapped_data) if x == 'Totales ']
+    total_thermal_generation = sum([mapped_data[i] for i in find_totals])
+    find_nuclear = [i+2 for i, x in enumerate(mapped_data) if x == 'nuclear']
+    total_nuclear_generation = sum([mapped_data[i] for i in find_nuclear])
+    #find_oil = [i+2 for i, x in enumerate(mapped_data) if x == 'oil']
+    #total_oil_generation = sum([mapped_data[i] for i in find_oil])
+
+    #Assume thermal generation is gas unless specifically mapped to another type.
+    total_gas_generation = total_thermal_generation - total_nuclear_generation
```

Comment:
I'll use wikipedia as a source.  There is only one functioning coal plant in Argentina right now with a capacity of 650MW.